tests/testsuite/cargo_add.rs

```diff
@@ -1504,6 +1504,74 @@ fn overwrite_inline_features() {
     );
 }
 
+#[cargo_test]
+fn overwrite_inherit_features_noop() {
+    init_registry();
+    let project_root =
+        project_from_template("tests/snapshots/add/overwrite_inherit_features_noop.in");
+    let cwd = &project_root;
+
+    cargo_command()
+        .env("__CARGO_TEST_CHANNEL_OVERRIDE_DO_NOT_USE_THIS", "nightly")
+        .arg("add")
+        .args(["foo", "-p", "bar"])
+        .current_dir(cwd)
+        .assert()
+        .success()
+        .stdout_matches_path("tests/snapshots/add/overwrite_inherit_features_noop.stdout")
+        .stderr_matches_path("tests/snapshots/add/overwrite_inherit_features_noop.stderr");
+
+    assert().subset_matches(
+        "tests/snapshots/add/overwrite_inherit_features_noop.out",
+        &project_root,
+    );
+}
+
+#[cargo_test]
+fn overwrite_inherit_noop() {
+    init_registry();
+    let project_root = project_from_template("tests/snapshots/add/overwrite_inherit_noop.in");
+    let cwd = &project_root;
+
+    cargo_command()
+        .env("__CARGO_TEST_CHANNEL_OVERRIDE_DO_NOT_USE_THIS", "nightly")
```

Comment:
#10581 will add a `masquerade_as_nightly_cargo()` function to `Command`.

Reply:
I saw that and put this in until that PR is merged or we plan to merge this one and then have a cleanup PR that fixes it